### ISSUE TITLE
Travis is stuck on "Removing @semantic-release"

### ISSUE THREAD

**Issue by DianaSuvorova** [travis-ci/travis-ci#8468]
Any suggestions on how to debug? 

https://travis-ci.org/DianaSuvorova/atom-karma-test-runner/builds/277965086

![preview](http://g.recordit.co/mhghgg3Jcj.gif)

**Comment by BanzaiMan** [comment]
Have you tested this build process on your machine?

There appears to be some bad interaction between your code (an/or its dependencies) and the script you are running (https://raw.githubusercontent.com/atom/ci/master/build-package.sh).

Notice that the script assumes that the environment variable `TRAVIS_OS_NAME` set, so if you want to test this on your development machine, you'd have to set it to `linux`.

From my brief testing, the script https://raw.githubusercontent.com/atom/ci/master/build-package.sh appears to run fine the first time. However, once the script completes and has installed modules in `node_modules`, the second invocation of it would result in the behavior you are describing.

The real solution probably lies with the "atom/ci" repository or the `apm` script it is provoking (`APM_SCRIPT_PATH="./atom/${ATOM_APP_NAME}/Contents/Resources/app/apm/node_modules/.bin/apm"`), but in the mean time, you can try disabling caching to force brand-new installation of Node.js dependencies to get your build past this problem.

**Comment by DianaSuvorova** [comment]
Thanks @BanzaiMan for looking into that! I did not test it locally.

Disabling the cache seemed to help with it being stuck on 'removing semantic-release'  step. Although I am unclear why would that be the case, and how did you come up with that solution given the symptoms?

The other problem as you correctly pointed out is with the apm script itself. To begin with it changes node version to 7 which semantic release does not support. But that completely separate  issue and def has nothing to do with travis.

I will close this ticket. I'd appreciate if you could give me some insight into your though process about disabling the cache.

Thanks,
Diana

**Comment by BanzaiMan** [comment]
> Although I am unclear why would that be the case, and how did you come up with that solution given the symptoms?

The initial objective would be to reproduce the problem, retracing the build process you are running in the build, using the Docker image we use. (https://docs.travis-ci.com/user/common-build-problems/#Troubleshooting-Locally-in-a-Docker-Image)

I ran the commands one by one, but the initial run failed with https://raw.githubusercontent.com/atom/ci/master/build-package.sh refusing to run. I looked into the script, noted how the script assumes the CI environment, then moved on. The initial did not reproduce the error, but then I noticed that your build was caching `node_modules`. I ran the test again to simulate that, and confirmed the error.

Once I have the reproduction, it was easy to confirm that the cache is the culprit; I removed `node_modules`, ran the script again, which succeeded. Ran the script again, and it failed.
